### PR TITLE
fix(content-picker): selection bug

### PR DESCRIPTION
**Pull request by priyajeet**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @priyajeet has signed the CLA. Thanks for the pull request!

File: src/elements/content-picker/ContentPicker.js [hunk 1]
```diff
@@ -102,7 +102,7 @@ type State = {
     isUploadModalOpen: boolean,
     rootName: string,
     searchQuery: string,
-    selected: { [string]: BoxItem },
+    selected: Array<BoxItem>,
```
Comment (jstoffan) [inline src/elements/content-picker/ContentPicker.js]: Ideally, we should have a single source of truth for the actual item objects. Can this be an array of item ids that we use to look up the item object from `currentCollection`, instead?

File: src/elements/content-picker/ContentPicker.js [hunk 2]
```diff
@@ -1152,7 +1137,7 @@ class ContentPicker extends Component<Props, State> {
         }: State = this.state;
         const { id, offset, permissions, totalCount }: Collection = currentCollection;
         const { can_upload }: BoxItemPermission = permissions || {};
-        const selectedCount: number = Object.keys(selected).length;
+        const selectedCount: number = selected.length;
```
Comment (jstoffan) [inline src/elements/content-picker/ContentPicker.js]: Should this use destructuring?

File: src/elements/content-picker/ItemList.js [hunk 3]
```diff
@@ -80,10 +84,17 @@ const ItemList = ({
             return '';
         }
 
-        const { selected, type } = items[index];
-        const isSelectable = isRowSelectable(selectableType, extensionsWhitelist, hasHitSelectionLimit, items[index]);
+        const { id: itemId, type } = items[index];
+        const isSelected = !!selected.find(sel => sel.id === itemId && sel.type === type);
```
Comment (jstoffan) [inline src/elements/content-picker/ItemList.js]: This logic seem to be duplicated in `onRowClick`

File: src/elements/content-picker/selectionCellRenderer.js [hunk 4]
```diff
@@ -15,11 +15,13 @@ export default (
     extensionsWhitelist: string[],
     hasHitSelectionLimit: boolean,
     isRadio: boolean,
+    selected: Array<BoxItem>,
```
Comment (jstoffan) [inline src/elements/content-picker/selectionCellRenderer.js]: Should we be passing the entire selected collection into each row? Can the parent pass in an `isSelected` value, instead?

File: src/elements/content-picker/selectionCellRenderer.js [hunk 5]
```diff
@@ -15,11 +15,13 @@ export default (
...
 ): (({ rowData: BoxItem }) => {}) => ({ rowData }: { rowData: BoxItem }) => {
-    const { name = '', selected = false } = rowData;
+    const { name = '', id, type } = rowData;
     const Component = isRadio ? RadioButton : Checkbox;
+    const isSelected = !!selected.find(sel => sel.id === id && sel.type === type);
```
Comment (jstoffan) [inline src/elements/content-picker/selectionCellRenderer.js]: Duplicated logic from the parent component?

**Pull request closed without merging**